app/assets/javascripts/controllers/ems_common/arbitration_profile_form_controller.js

```diff
@@ -0,0 +1,101 @@
+        $scope.arbitrationProfileModel.description          = arbitrationProfileData.description;
+        $scope.arbitrationProfileModel.authentication_id    = convertToString(arbitrationProfileData.authentication_id);
+        $scope.arbitrationProfileModel.availability_zone_id = convertToString(arbitrationProfileData.availability_zone_id);
+        $scope.arbitrationProfileModel.cloud_network_id     = convertToString(arbitrationProfileData.cloud_network_id);
+        $scope.arbitrationProfileModel.cloud_subnet_id      = convertToString(arbitrationProfileData.cloud_subnet_id);
+        $scope.arbitrationProfileModel.flavor_id            = convertToString(arbitrationProfileData.flavor_id);
+        $scope.arbitrationProfileModel.security_group_id    = convertToString(arbitrationProfileData.security_group_id);
+      }
+
+      $scope.arbitrationProfileModel.authentications    = arbitrationProfileOptions.authentications
+      $scope.arbitrationProfileModel.availability_zones = arbitrationProfileOptions.availability_zones
+      $scope.arbitrationProfileModel.cloud_networks     = arbitrationProfileOptions.cloud_networks
+      $scope.arbitrationProfileModel.cloud_subnets      = arbitrationProfileOptions.cloud_subnets
+      $scope.arbitrationProfileModel.flavors            = arbitrationProfileOptions.flavors
+      $scope.arbitrationProfileModel.security_groups    = arbitrationProfileOptions.security_groups
+
+      $scope.modelCopy = angular.copy( $scope.arbitrationProfileModel );
+    };
+
+  $scope.cancelClicked = function() {
+    var task = $scope.newRecord ? "Add" : "Edit"
+    var msg = sprintf(__(task + " of Arbitration Profile %s was cancelled by the user"), $scope.arbitrationProfileModel.description);
+    postService.cancelOperation(redirectUrl, msg);
+    $scope.angularForm.$setPristine(true);
+  };
+
+  $scope.resetClicked = function() {
+    $scope.arbitrationProfileModel = angular.copy( $scope.modelCopy );
+    $scope.angularForm.$setUntouched(true);
+    $scope.angularForm.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  $scope.saveClicked = function() {
+    var successMsg = sprintf(__("Arbitration Profile %s was saved"), $scope.arbitrationProfileModel.name);
+    postService.saveRecord('/api/arbitration_profiles/' + arbitrationProfileFormId,
+      redirectUrl,
+      setProfileOptions(),
+      successMsg);
+    $scope.angularForm.$setPristine(true);
+  };
+
+  $scope.addClicked = function($event, formSubmit) {
+    var successMsg = sprintf(__("Arbitration Profile %s was added"), $scope.arbitrationProfileModel.name);
+    postService.createRecord('/api/arbitration_profiles',
+      redirectUrl,
+      setProfileOptions(),
+      successMsg);
+    $scope.angularForm.$setPristine(true);
+  };
+
+  // extract ems_id from url
+  var emsId = (/ems_cloud\/arbitration_profile_edit\/(\d+)/.exec($location.absUrl())[1]);
```

Comment:
Agreed, that's why I figure it should throw .. but then again, urls with `undefined` tend to be relatively noticeable..